### PR TITLE
feat: support WalletConnect dApps in QR Code button

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Brings support for `wc:` urn parsing in the QR code button, leveraging our existing QR code logic.
This is done through passing `isOpen` and `handleClose` from the QR code modal to `<ConnectModal />`, effectively binding it to the QR code modal opened/closed state, and passing the decoded QR value as an `initialUri` prop, to be used `uri` form field.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Current "WalletConnect dApps" button could be broken. Note that it is currently a bit flaky and doesn't like re-connecting to a session when the session was closed by closing the tab as opposed to closing disconnect. This is a current prod issue. Always click disconnect from the WalletConnect dApps button's menu before re-initiating a connection while testing this, and make sure to always hard refresh your App and generate a new QR as well as refreshing the ShapeShift app before re-initiating a connection, both on prod and in this diff while testing it.

## Testing

- WalletConnect dApps button shows no regressions
- QR code button handles `wc:` WalletConnect dApps URLs gracefully
- Ensure this looks and feels good on mobile. WalletConnect is the most popular way to connect on mobile, and the QR code button is eventually going to replace the current WalletConnect dApps one so we want to ensure its feature parity

### Engineering

- The branching between send/QR code logic/rendering in the QR code Form doesn't look too ugly

### Operations

- ☝🏽

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4406** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4406" target="_blank">Graphite</a> 
    * **PR #4407** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4407" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4407?utm_source=stack-comment).

File: src/plugins/walletConnectToDapps/components/modals/connect/ConnectContent.tsx [hunk 1]
```diff
@@ -50,15 +54,17 @@ export const ConnectContent: React.FC<ConnectContentProps> = ({ handleConnect })
 
   const { register, handleSubmit, control, formState, setValue } = useForm<FormValues>({
     mode: 'onChange',
-    defaultValues: { uri: '' },
+    defaultValues: { uri: initialUri },
   })
+
```
Comment (gomesalexandre) [inline src/plugins/walletConnectToDapps/components/modals/connect/ConnectContent.tsx:59]:
```suggestion
```

File: src/plugins/walletConnectToDapps/components/modals/connect/ConnectContent.tsx [hunk 2]
```diff
@@ -50,15 +54,17 @@ export const ConnectContent: React.FC<ConnectContentProps> = ({ handleConnect })
...
   const handleQrScanSuccess = useCallback(
     (uri: string) => {
       setValue('uri', uri)
       toggleQrCodeView()
     },
     [setValue, toggleQrCodeView],
   )
+
```
Comment (gomesalexandre) [inline src/plugins/walletConnectToDapps/components/modals/connect/ConnectContent.tsx:67]:
```suggestion
```

File: src/components/Modals/QrCode/Form.tsx [hunk 3]
```diff
@@ -88,6 +95,10 @@ export const Form: React.FC<QrCodeFormProps> = ({ accountId }) => {
   const handleQrSuccess = useCallback(
     (decodedText: string) => {
       ;(async () => {
+        // If this is a WalletConnect dApp QR Code, skip the whole send logic and render the QR Code Modal instead.
+        // There's no need for any RFC-3986 decoding here since we don't really care about parsing and WC will do that for us
+        if (decodedText.startsWith('wc:')) return setWalletConnectDappUrl(decodedText)
```
Comment (0xdef1cafe) [inline src/components/Modals/QrCode/Form.tsx:100]: will this work with v2?

Comment (gomesalexandre) [reply src/components/Modals/QrCode/Form.tsx:100]: Yeah they use the same `wc:` URN scheme as per [EIP-1328](https://eips.ethereum.org/EIPS/eip-1328), only the `version` param differs. I have tested with a bunch of different dApps (Hop, 1Inch, Safe, Uniswap), and just re-tested as a sanity check with https://react-dapp-v2-with-web3js.vercel.app/

image

**Review by 0xdef1cafe — APPROVED** [review]
untested but will let ops test

**Pull request merged**